packages/hooks/src/hooks/useCopyToClipboard.ts

```diff
@@ -0,0 +1,31 @@
+import { useState } from 'react';
+
+type CopiedValue = string | null;
+type CopyFn = (text: string) => Promise<boolean>; // Return success
+
+export function useCopyToClipboard(): [CopiedValue, CopyFn] {
+	const [copiedText, setCopiedText] = useState<CopiedValue>(null);
+
+	const copy: CopyFn = async text => {
+		if (!navigator?.clipboard) {
+			console.warn('Clipboard not supported');
+			return false;
+		}
+
+		try {
+			await navigator.clipboard.writeText(text);
+			if (text !== copiedText) {
```

Comment:
not sure this condition is required, because react shouldn't trigger re-render if you set state with the same primitive value.